### DISCUSSION TITLE
[Feature request] option too choose web browser

### DISCUSSION THREAD

**Discussion by jimblue** [webpack-contrib/webpack-bundle-analyzer#565] [Ideas]
Hi,

Actually `webpack-bundle-analyzer` doesn't offer an option to choose which browser open.

The option can only be set to default or null.

As many people use a different browser for development than for browsing it could be nice to have such choice.

Cheers

**Comment by jimblue** [comment]
Hey,

Any update about this?

Cheers!

**Comment by valscion** [comment]
The underlying package we use is [`opener`](https://github.com/domenic/opener) and it will open the default browser you've configured.

I'd like to avoid adding a new configuration option for the browser, so I'm hesitant on adding this feature.

**Comment by jimblue** [comment]
Hey!

Why not? It will not break anything, only add flexibility, which is always good 😄!

Consider that peoples often use two browsers, it would be a nice feature to add:
- the default OS browser for everyday use to keep deep integration with the system
- google chrome browser for development (the more often)

Cheers

**Comment by jimblue** [comment]
It could be also useful for developer using **webpack-bundle-analyzer** in their library...

**Comment by jimblue** [comment]
To make it even better, I'll suggest to replace **opener** with **[open](https://github.com/sindresorhus/open#readme)**.
This package have more stars and is better maintain.
Also the API is way better. it has much more options and it return a promise when the opened app exit. 😃

**Comment by valscion** [comment]
> which is always good

Additional flexibility is a double-edged sword. It makes the API surface larger, making the tool harder to use as documentation grows longer. Adding new options also introduces maintenance burden as we will need to make sure the new feature does not accidentally break.

We don't currently have the testing infrastructure for the `openAnalyzer` flag, and adding more features like that will make it much more likely for us to break it in the future.

So that is why I am not eagerly adding this feature.

**Comment by jimblue** [comment]
> Additional flexibility is a double-edged sword

I would be agree if it was a huge add, but it's just easy as passing **browser** argument from **webpack-bundle-analyzer** to **open**.

How could that break anything or make future that more complicated?

I don't get it honestly...

**Comment by sindresorhus** [comment]
I would probably wait until https://github.com/sindresorhus/open/issues/116 is implemented.

**Comment by valscion** [comment]
Thanks @sindresorhus, looks like that would indeed help.

**Comment by DisposaBoy** [comment]
FWIW: Until/if this is supported, you can hack around it with a script that overrides `xdg-open` or `open` on Linux and Mac.

e.g. I put this script named `xdg-open` in a directory that comes before `/usr/bin` in `$PATH`.

	#!/bin/sh

	if test -n "$XDG_OPEN"; then
		$XDG_OPEN "$@"
	else
		/usr/bin/xdg-open "$@"
	fi

And it allows me to choose my browser in the `package.json` `script` e.g. `XDG_OPEN=browser ...webpack-build-command...`.

**Comment by jimblue** [comment]
Hi there,
Just a friendly ping as the dependency issue seems close now : https://github.com/sindresorhus/open/issues/116
Is this issue status is still blocked ?

**Comment by valscion** [comment]
Might not be, maybe a dependency bump is needed with this feature. Feel free to create a PR :)
